### PR TITLE
refactor(executor): refactor queries executor to prepare

### PR DESCRIPTION
**Pull request by dqhl76**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

**This PR should not affect currently used executor**

1. Move the logic related to specific queries from QueriesPipelineExecutor to PipelineExecutor
2. Add functionality to Graph for interrupting its execution; Executor should not terminate after Graph completion
3. Modify the logic for creating Graph in PipelineExecutor to prepare for future conversion to Server/Client mode

**Not ready to review. Still work in progress:**
1. refine error handle

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14965)

### CODE REVIEW

**Comment by dqhl76** [comment]
I will clean some debug info latter and disable the new executor. For next PR, I will do the following things:
1. change the exeuctor creat method
2. add time limit abort (move the aborting monitor from `QueriesPipelineExeuctor` to `PipelineExeuctor`)

File: src/query/service/src/pipelines/executor/executor_graph.rs [hunk 1]
```diff
@@ -711,11 +751,57 @@ impl RunningGraph {
         }
     }
 
+    /// Checks if all nodes in the graph are finished.
+    pub fn is_all_nodes_finished(&self) -> bool {
+        info!(
```
[the remark this answers is not in the document]
Comment (dqhl76) [reply src/query/service/src/pipelines/executor/executor_graph.rs]: resolved

File: src/query/service/src/pipelines/executor/queries_pipeline_executor.rs [hunk 2]
```diff
@@ -413,16 +182,30 @@ impl QueriesPipelineExecutor {
...
+                            );
+                        }
+                        if graph.is_should_finish() {
+                            // TODO: temporary finish method, will remove after change executor to a global service
+                            self.finish(None);
+                        }
+                        if graph.is_all_nodes_finished() {
+                            graph.should_finish(Ok(()))?;
+                        }
+                    }
```
Comment (dqhl76) [inline src/query/service/src/pipelines/executor/queries_pipeline_executor.rs:204]: I am not confident about this part. /cc @zhang2014 
I tried to separate `query's error` and `executor's error`. Do you have any comment?

File: src/query/service/src/pipelines/executor/queries_pipeline_executor.rs [hunk 3]
```diff
@@ -413,16 +182,30 @@ impl QueriesPipelineExecutor {
...
-                            &self.global_tasks_queue,
-                            &mut context,
-                            self,
-                        );
+                match context.execute_task(Some(self)) {
+                    Ok(Some((executed_pid, graph))) => {
+                        // Not scheduled graph if pipeline is finished.
+                        if !self.global_tasks_queue.is_finished() && !graph.is_should_finish() {
+                            // We immediately schedule the processor again.
+                            let schedule_queue = graph.clone().schedule_queue(executed_pid)?;
```
Comment (zhang2014) [inline src/query/service/src/pipelines/executor/queries_pipeline_executor.rs:190]: schedule_queue is also query error(Processor::event).

**Review by zhang2014 — APPROVED** [review]

**Pull request merged**